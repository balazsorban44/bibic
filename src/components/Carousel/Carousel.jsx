import React from "react"

import Slide from "react-reveal/Slide"

import Loading from "ui/Loading"
import useScrollTo from "hooks/useScrollTo"

import CarouselWrapper from "./CarouselWrapper"
import DefaultCarouselItem from "./CarouselItem"
import useGallery from "hooks/data/useGallery"

export const Carousel = ({type, title, subtitle, className, item, slider=true, count}) => {
  const CarouselItem = item || DefaultCarouselItem

  useScrollTo(0,0)

  const [gallery] = useGallery(type)


  const children =
    Object.values(gallery)
      .slice(0, count)
      .map((item, key) =>
        slider ?
          <Slide key={key} right>
            <CarouselItem className="carousel-item" {...item} />
          </Slide> :
          <CarouselItem className="carousel-item" key={key} {...item} />
      )

  return (
    <div className={className || "carousel-wrapper"} id={type}>
      {title ?
        <div className="title">
          <h2>{title}</h2>
        </div> : null
      }
      {children ?
        slider ?
          <CarouselWrapper defaultWait={10000}>
            {children}
          </CarouselWrapper> :
          children :
        // REVIEW:  turn to component
        <div style={{display: "flex", alignItems: "center", justifyContent: "center"}}>
          <Loading/>
        </div>
      }
      {subtitle ? <h3>{subtitle}</h3> : null }
    </div>
  )
}

export default Carousel

