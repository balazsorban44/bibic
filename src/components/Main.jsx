import React from "react"
import Sunflower from "modules/Sunflower"
import Services from "modules/Facilities"
import Rooms from "modules/Rooms"
import Prices from "modules/Prices"
import Feedbacks from "modules/Feedbacks"
import Footer from "modules/Footer"

export default () =>
  <>
    <Sunflower/>
    <Services/>
    <Rooms/>
    <Prices/>
    <Feedbacks/>
    <Footer/>
  </>
