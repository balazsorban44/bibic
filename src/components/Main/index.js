import React, {Component} from 'react'
import Introduction from './components/Introduction'
import Sunflower from './components/Sunflower'
import Services from './components/Services'
import Rooms from './components/Rooms'
import Prices from './components/Prices'
import Slider from '../shared/Slider'
import Foods from './components/Foods'

import {Element} from 'react-scroll'


export default class Main extends Component {
  render() {
    return (
      <main>
        <Element name="bemutatkozas">
          <Introduction/>
        </Element> 
        <Element name="napraforgo">
          <Sunflower/>
        </Element>
        <Element name="szolgaltatasok">
          <Services/>
        </Element> 
        <Element name="szobak">
          <Rooms/>
        </Element>
        <Element name="arak">
          <Prices/>
        </Element> 
        <Element name="etelek">
          <Foods databaseRef="foods"/>
        </Element>
        <Element name="rendezvenyek">
          <Slider
            title="RendezvÃ©nyek"
            sectionId="rendezvenyek"
            databaseRef="events"
          />
        </Element>  
        <Element name="programok">
          <Slider
            title="Programok"
            sectionId="programok"
            databaseRef="programs"
            alignRight
          />
        </Element> 
      </main>
    )
  }
}