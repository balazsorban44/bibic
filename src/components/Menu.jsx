import React, {
  lazy, Suspense, Component
} from 'react'
import {Link as RouteLink, withRouter, useHistory} from 'react-router-dom'
import {Link} from 'react-scroll'
import bibic from '../assets/icons/bibic.png'
import logo from '../assets/icons/logo.png'
import Fade from "react-reveal/Fade"
import {BASE_URL} from '../utils/constants'
import {withTranslation, useTranslation} from 'react-i18next'
import LanguageSelector from './LanguageSelector'

const RoomMenu = lazy(() => import("./RoomMenu"))


export class Menu extends Component {

  state = {
    isMenuOpen: false,
    isScrolled: false,
    isRoomMenuOpen: true,
    menu: [
      {name: "intro", to:"bemutatkozas"},
      {name: "facilities", to:"szolgaltatasok"},
      {name: "rooms", to:"szobak"},
      {name: "prices", to:"arak"},
      {
        name: "menu", to:"etelek", component: RouteLink
      },
      {
        name: "events", to:"rendezvenyek", component: RouteLink
      },
      {name: "feedback", to:"visszajelzesek"},
      {name: "contact", to:"kapcsolat"}
    ]
  }

  componentDidMount() {
    window.addEventListener("scroll", this.isScrolled, false)
    window.addEventListener("resize", this.isResized, false)
    this.isScrolled()
    this.isResized()
    if (window.innerWidth > 768) this.setState({isRoomMenuOpen: false})
  }

  componentWillUnmount() {
    window.removeEventListener("scroll", this.isScrolled, false)
    window.removeEventListener("resize", this.isResized, false)
  }

  isScrolled = () => this.setState({isScrolled: window.pageYOffset >= 100})

  isResized = () => this.setState({width: window.innerWidth})

  handleShowRoomMenu = () => this.setState({isRoomMenuOpen: true})

  handleHideRoomMenu = () => this.setState({isRoomMenuOpen: false})

  handleMenuToggle = () => this.setState(({isMenuOpen}) => ({isMenuOpen: !isMenuOpen}))

  handleHideMenu = () => this.setState({isMenuOpen: false, isRoomMenuOpen: false})

  render() {
    const {
      isMenuOpen, isScrolled, isRoomMenuOpen, width, menu
    } = this.state
    const isBigScreen = width > 768
    return (
      <Fade down>
        <div
          className={`menu ${isScrolled ? "menu-scrolled" : ""} ${isMenuOpen ? "" : "menu-hidden"}`}
        >
          <Fade right>
            <div
              className={`hamburger ${isMenuOpen ? "hamburger-clicked" : ""}`}
              onClick={this.handleMenuToggle}
            ><span/><span/><span/></div>
          </Fade>
          <a
            className={`menu-logo ${isScrolled ? "menu-logo-scrolled" : ""}`}
            href={BASE_URL}
          >
            <picture>
              <source
                media="(min-aspect-ratio: 7/6) and (min-width: 1024px)"
                srcSet={logo}
              />
              <img
                alt="logo"
                src={bibic}
              />
            </picture>
          </a>
          <nav>
            <Fade
              cascade
              down
            >
              <ul>
                {menu.map(({
                  name, to, component: Component=Link, offset
                }) =>
                  <li key={to}>
                    <Component
                      href={`#${to}`}
                      offset={isBigScreen ? offset !== undefined ? offset : -64 : 0}
                      onClick={this.handleHideMenu}
                      onMouseEnter={() => to==="szobak" && this.handleShowRoomMenu()}
                      smooth={Component === Link ? true : undefined}
                      to={to}
                    >
                      {this.props.t(`menu.${name}`)}
                    </Component>
                  </li>
                )}
                <LanguageSelector/>
              </ul>
            </Fade>
          </nav>
          <Suspense fallback={null}>
            <ul className={`room-menu ${isRoomMenuOpen ? "room-menu-show" : ""}`}>
              <RoomMenu
                isBigScreen={isBigScreen}
                onClick={this.handleHideMenu}
              />
            </ul>
          </Suspense>
        </div>
      </Fade>
    )
  }
}

export default withTranslation()(Menu)


export const BackMenu = () => {
  const {goBack, location: {search}} = useHistory()
  const customReturn = new URLSearchParams(search).get("backTo")
  const [t] = useTranslation()
  const title = customReturn || t("menu.homepage")

  if (customReturn) {
    return (
      <button
        className="back-to-home"
        onClick={goBack}
        title={title}
      >
        <Title title={title}/>
      </button>
    )
  } else {
    return (
      <RouteLink
        className="back-to-home"
        title={title}
        to="/"
      >
        <Title title={title}/>
      </RouteLink>
    )
  }
}

const Title = ({title}) => {
  const [t] = useTranslation()
  return (
    <>
      {"‚Üê"}
      <span>
        {t("menu.back-to", {title})}
      </span>
    </>
  )
}