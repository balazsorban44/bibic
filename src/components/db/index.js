import moment from 'moment'
import QueryString from 'query-string'
import React, {Component, createContext} from 'react'
import {withRouter} from 'react-router-dom'
import {isQueryString, translate} from '../../utils/language'
import {
  submitReservation, getPrice, normalizeReservation
} from './reservation'
import {valueToState} from '../../utils/validate'
import {isEquivalent} from '../../utils/compare'
import {
  CLOUD_FUNCTION_BASE_URL, TODAY, TOMORROW
} from '../../utils/constants'

import {sendNotification} from "./notification"
import {submitMessage} from './message'
import {submitFeedback} from './feedback'
import {FEEDBACKS_REF} from '../../lib/firebase'

const Store = createContext()
/**
 * Makes the Store values available
 * @param {Component} WrappedComponent The component to pass the store values to
 * @returns {Component} Component with the Store values
 */
export const withStore = WrappedComponent =>
  class extends Component {
    render() {
      return (
        <Store.Consumer>
          {values =>
            <WrappedComponent
              {...{...values,
                ...this.props}}
            />
          }
        </Store.Consumer>
      )
    }
  }


const initialReservation = {
  roomId: null,
  from: TOMORROW.toDate(),
  to: TOMORROW.toDate(),
  name: "",
  email: "",
  address: "",
  tel: "",
  message: "",
  adults: 1,
  children: [],
  foodService: "breakfast",
  price: 0
}

const initialMessage = {
  content: "",
  subject: "other",
  address: "",
  name: "",
  email: "",
  tel: ""
}

export class Database extends Component {

  state = {
    hero: [],
    isReserving: false,
    isMessageLoading: false,
    tomorrow: TOMORROW,
    paragraphs: {},
    galleries: {},
    month: TODAY,
    reservation: initialReservation,
    message: initialMessage,
    rooms: [],
    roomServices: [],
    overlaps: [],
    feedbacks: {all: [], rooms: []}
  }


  // Fetch all initial data from Firebase.
  componentDidMount() {

    Promise.all([
      import("../../assets/images/hero/1.jpg"),
      import("../../assets/images/hero/2.jpg")
    ]).then(images => {
      this.setState({hero: images.map(({default: img}) => img)})
    })

    import("../../lib/firebase").then(({
      PARAGRAPHS_REF, ROOMS_REF, ROOM_SERVICES_REF, GALLERIES_REF, FEEDBACKS_FS_REF
    }) => {

      FEEDBACKS_FS_REF.where("accepted", "==", true).orderBy("timestamp", "asc").limit(20).onSnapshot(data => {
        const feedbacks = []
        data.forEach(feedback => {
          feedbacks.push(feedback.data())
        })
        this.setState(({feedbacks: prevFeedbacks}) => ({feedbacks: {...prevFeedbacks, all: feedbacks}}))
      })

      FEEDBACKS_REF.on("value", snap =>
        this.setState(
          ({feedbacks: prevFeedbacks}) => ({feedbacks: {...prevFeedbacks, rooms: snap.val()}})
        )
      )

      PARAGRAPHS_REF
        .on("value", snap => {
          const paragraphs = {}
          Object.entries(snap.val()).forEach(([paragraphType, paragraphList]) => {
            paragraphs[paragraphType] = Object.values(paragraphList)
              .sort((a, b) => a.order - b.order)
          })
          this.setState({paragraphs})
        })

      GALLERIES_REF
        .on("value", snap => {
          const galleries = {}
          Object.entries(snap.val()).forEach(([galleryType, galleryList]) => {
            galleries[galleryType] = Object.values(galleryList)
              .sort((a, b) => a.order - b.order)
          })
          this.setState({galleries})
        })


      ROOM_SERVICES_REF
        .on("value", snap =>
          this.setState({roomServices: Object.entries(snap.val())})
        )


      ROOMS_REF.once("value", snap => this.setState({rooms: snap.val()}))
        .then(() => this.updateByURL(this.props.location.search, true))
    })

  }


  componentDidUpdate({location: {search: prevSearch}}, {reservation: prevReservation}) {
    const {search} = this.props.location
    const {reservation} = this.state

    if (!isEquivalent(prevReservation, reservation)) this.updatePrice()
    if (prevSearch !== search) {
      this.updateByURL(search)
      this.updatePrice()
    }
  }


  /*
   * ----------------------------------------------------------------------------
   * Reservation
   */

  handleSubmitReservation = () =>
    submitReservation(
      normalizeReservation(this.state.reservation),
      isReserving => this.setState({isReserving}),
      () => this.setState({reservation: initialReservation}),
      () => this.props.history.push(""),
      this.state.rooms
    )

  updatePrice = () => {
    const {rooms, reservation:{roomId}} = this.state
    const room = rooms.find(room => room.id === roomId)

    this.setState(({reservation}) => ({reservation: {...reservation,
      price: room ? getPrice(room, reservation) : 0}}))
  }

  /**
   * Updates the reservation either in the state or in the URL
   * If the key is not private, it adds the key=value pair to the URL
   * @param {string} key - reservation key
   * @param value - reservation value
   */
  updateReservation = (key, value) => {
    if (isQueryString(key)) {
      const {history} = this.props
      const search = QueryString.parse(history.location.search)
      search[translate(key)] = key === "foodService" ? translate(value) : value
      history.push(`foglalas?${ QueryString.stringify(search)}`)
    } else this.setState(({reservation}) => ({reservation: {...reservation,
      [key]: value}}))
  }

  fetchOverlaps = async () => {
    try {
      let overlaps = await fetch(`${CLOUD_FUNCTION_BASE_URL}/getOverlaps?roomId=${this.state.reservation.roomId}`)
      overlaps = await overlaps.json()
      this.setState({overlaps: overlaps.map(({start, end}) => moment.range(start, end))})
    } catch (error) {
      sendNotification("error", error.message)
    }
  }


  /*
   * ----------------------------------------------------------------------------
   * Message
   */

  handleSubmitMessage = () =>
    submitMessage(
      {...this.state.message},
      isMessageLoading => this.setState({isMessageLoading}),
      () => this.setState({message: initialMessage}),
      () => this.props.history.push("")
    )

  updateMessage = (key, value) => {
    if (isQueryString(key)) {
      const {history} = this.props
      const search = QueryString.parse(history.location.search)
      search[translate(key)] = key === "subject" ? translate(value) : value
      history.push(`uzenet?${ QueryString.stringify(search)}`)
    } else this.setState(({message}) => ({message: {...message,
      [key]: value}}))
  }


  /*
   * ----------------------------------------------------------------------------
   * Feedback
   */

  handleSubmitFeedback = feedback =>
    submitFeedback(feedback, loading => this.setState({loading}))


  /*
   * ----------------------------------------------------------------------------
   * Routing
   */


  /**
   * Updates the state from the URL
   * @param {object} values
   * @param {boolean} [isInitial=false]
   */
  updateByURL = (queryString, isInitial=false) => {
    if (queryString) {
      const reservation = {...this.state.reservation}
      const message = {...this.state.message}
      queryString = QueryString.parse(queryString)
      Object.entries(queryString).forEach(([key, value]) => {
        key = translate(key)
        value = key === "foodService" || key === "subject" ? translate(value) : value
        if (this.props.location.pathname.replace("/", "") === "uzenet") {
          message[key] = valueToState(key, value)
        } else {
          reservation[key] = valueToState(key, value)
        }
      })

      this.setState({reservation, message}, () => {
        let {reservation: {from}} = this.state
        if (isInitial) {
          from = moment(from)
          this.setState({month: from})
        }
      })
    }
  }


  render() {
    return (
      <Store.Provider
        value={{
          submitReservation: this.handleSubmitReservation,
          submitMessage: this.handleSubmitMessage,
          submitFeedback: this.handleSubmitFeedback,
          updateReservation: this.updateReservation,
          updateMessage: this.updateMessage,
          fetchOverlaps: this.fetchOverlaps,
          ...this.state
        }}
      >
        {this.props.children}
      </Store.Provider>
    )
  }
}


export default withRouter(Database)
