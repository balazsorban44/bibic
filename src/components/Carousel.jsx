import React, {Component, Fragment} from 'react'
import Gallery from './Gallery'
import withRouter from 'react-router/withRouter'
import {BackMenu} from '../Menu'
import {Next, Prev} from './Elements'
import Carousel from 'nuka-carousel'


class Carousel extends Component {

  componentDidMount() {
    window.scrollTo(0, 0)
  }

  render() {
    let {path} = this.props.match
    path = path.replace("/", "")
    return(
      <Fragment>
        <BackMenu/>
        <section id={path}>
          <h2>Ã‰teleink</h2>
          <Gallery
            component={Carousel}
            componentProps={{
              // autoplay: true,
              wrapAround: true,
              className: "foods",
              heightMode: "max",
              cellSpacing: 40,
              renderCenterRightControls: Next,
              renderCenterLeftControls: Prev,
              renderTopRightControls: Next,
              renderTopLeftControls: Prev
            }}
            item={FoodItem}
            path={path}
          />
        </section>
      </Fragment>
    )}
}

export default withRouter(Carousel)


const FoodItem = ({
  title, desc, SIZE_640, SIZE_1280, SIZE_1440, SIZE_ORIGINAL
}) =>
  <div className="food">
    <picture>
      <source
        media="(max-width: 640px)"
        srcSet={SIZE_640}
      />
      <source
        media="(max-width: 1280px)"
        srcSet={SIZE_1280}
      />
      <source
        media="(max-width: 1920px)"
        srcSet={SIZE_1440}
      />
      <img
        alt={title}
        src={SIZE_ORIGINAL}
      />
    </picture>
    <div className="food-desc">
      <h4>{title}</h4>
      <p>{desc}</p>
    </div>
  </div>