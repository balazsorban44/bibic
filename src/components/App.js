import React, {lazy, Suspense} from 'react'
import {Route, Switch} from 'react-router-dom'

import {BackMenu} from './Menu'
import ChatFAB from './ChatFAB'
import {routes} from '../utils/constants'
import Top from "./Top"
import NotFound from "./NotFound"
import {Loading} from './shared/Elements'
import Footer from './Footer'


/**
 * NOTE: Asynchronously fetching Components
 * that do not need to be loaded at startup
 */

import ReservationForm from "./ReservationForm"
import FeedbackForm from "./FeedbackForm"
import MessageForm from "./MessageForm"
import Carousel from "./shared/Carousel"
import MoreServices from "./MoreServices"
import {useTranslation} from 'react-i18next'

const Main = lazy(() => import("./Main"))


const App = () => {
  const [t] = useTranslation()
  return (
    <>
      <Route render={({location: {pathname}}) => pathname !== routes.HOME && <BackMenu/>} />
      <Switch>
        <Route
          exact
          path={routes.HOME}
          render={() =>
            <>
              <Top/>
              <ChatFAB/>
              <Suspense fallback={<Loading/>}>
                <Main/>
              </Suspense>
            </>
          }
        />
        <Route
          component={() =>
            <Carousel
              className="foods"
              itemClassName="food"
              subtitle={t("foods.subtitle")}
              title={t("foods.title")}
            />
          }
          exact
          path={routes.FOODS}
        />
        <Route
          component={() =>
            <Carousel
              className="events"
              itemClassName="event"
              title={t("events.title")}
            />}
          exact
          path={routes.EVENTS}
        />
        <Route
          component={ReservationForm}
          exact
          path={routes.RESERVE}
        />
        <Route
          component={MessageForm}
          path={routes.MESSAGE}
        />
        <Route
          component={FeedbackForm}
          path={routes.FEEDBACK_FORM}
        />
        <Route
          component={MoreServices}
          path={routes.SERVICES}
        />
        <Route component={NotFound}/>
      </Switch>
      <Footer/>
    </>
  )
}


export default App
