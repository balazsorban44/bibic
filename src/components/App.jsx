import React, {lazy} from "react"
import {Route, Switch} from "react-router-dom"
import {useTranslation} from "react-i18next"
import Loadable from "react-loadable"

import routes from "utils/routes"

import Landing from "modules/Landing"
import Introduction from "modules/Introduction"
import Providers from "modules/context"

import {ChatFAB} from "components/Pinned"
import BackButton from "components/BackButton"
import NotFound from "components/NotFound"
import Loading from "components/Loading"
import withLazy from "components/withLazy"


/**
 * NOTE: Asynchronously fetching Components
 * that do not need to be loaded at startup
 */

const Reservation = Loadable({
  loader: () => import("modules/reservation"),
  loading: Loading
})
const MessageForm = Loadable({
  loader: () => import("modules/Message"),
  loading: Loading
})
const FeedbackForm = Loadable({
  loader: () => import("modules/FeedbackForm"),
  loading: Loading
})

const ServicesPage = Loadable({
  loader: () => import("modules/Services/ServicesPage"),
  loading: Loading
})

const Carousel = Loadable({
  loader: () => import("components/Carousel"),
  loading: Loading
})

const Main = withLazy(lazy(() => import("components/Main")))

function App () {
  const [t] = useTranslation("common")
  return (
    <Providers>
      {/* <BetaWarning/> */}
      <Route render={({location: {pathname}}) => pathname !== routes.HOME && <BackButton/>} />
      <Switch>
        <Route
          exact
          path={routes.HOME}
          render={() =>
            <>
              <Landing/>
              <Introduction/>
              <ChatFAB/>
              <Main/>
            </>
          }
        />
        <Route
          component={() =>
            <Carousel
              itemClassName="food"
              subtitle={t("subtitles.foods")}
              title={t("titles.foods")}
              type="etelek"
            />
          }
          exact
          path={routes.FOODS}
        />
        <Route
          component={() =>
            <Carousel
              itemClassName="event"
              title={t("titles.events")}
              type="rendezvenyek"
            />}
          exact
          path={routes.EVENTS}
        />
        <Route
          component={MessageForm}
          path={routes.MESSAGE}
        />
        <Route
          component={Reservation}
          exact
          path={routes.RESERVE}
        />
        <Route
          component={FeedbackForm}
          path={routes.FEEDBACK_FORM}
        />
        <Route
          component={ServicesPage}
          path={routes.SERVICES}
        />
        <Route component={NotFound}/>
      </Switch>
    </Providers>
  )
}

export default App