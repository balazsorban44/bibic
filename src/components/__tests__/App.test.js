import App from "../App"
import {BackMenu} from "../Menu"

import {MemoryRouter} from "react-router"
import {Database} from "../db"
import NotFound from "../NotFound"
import MoreServices from "../MoreServices"
import {Carousel} from "../shared/Carousel"
import Link from "react-router-dom/Link"
import Main from "../Main"

global.scrollTo = jest.fn()

const renderRoutes = path =>
  mount(
    <MemoryRouter initialEntries={[path]}>
      <Database hero={[]}>
        <App/>
      </Database>
    </MemoryRouter>
  )

describe("App component", () => {
  const wrapper = shallow(<App/>)

  it("renders correctly", () => {
    expect(wrapper).toHaveLength(1)
  })

  describe("Back button", () => {
    it("show if not on home page", () => {
      const component = renderRoutes("/random")
      expect(component.find(BackMenu).length).toBe(1)
    })


    it.skip("don't show on home page", () => {
      const component = renderRoutes("/")
      expect(component.find(BackMenu).length).toBe(0)
    })
  })

  describe("Not found", () => {
    it("renders correctly", () => {
      const component = renderRoutes("/a")
      expect(component.find(Link).length).toBe(2)
      expect(component.find(Link).first().prop("to")).toBe("/")
      expect(component.find(Link).last().prop("to")).toBe("/")
    })
  })

  describe("MoreServices", () => {
    it("renders correctly", () => {
      const component = renderRoutes("/szolgaltatasok")
      expect(component.find(MoreServices).length).toBe(1)
    })
  })

  describe("Foods", () => {
    it("renders correctly", () => {
      const component = renderRoutes("/etelek")
      expect(component.find(Carousel).length).toBe(1)
    })
  })

})


describe("Main component", () => {
  const wrapper = shallow(<Main/>)

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot()
  })
})