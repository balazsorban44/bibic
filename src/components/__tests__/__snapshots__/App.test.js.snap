// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component renders correctly 1`] = `ShallowWrapper {}`;
